File: reporting-test-client/IrrigationReportingWebApi/IrrigationReportingWebApi/Models/IrrigationEventsResponse.cs
using System;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using System.Linq;

namespace Trimble.Ag.IrrigationReporting.IrrigationReportingWebApi.Models
{
	public class IrrigationEventsResponse
	{
		public IEnumerable<IrrigationEventSummary> Summary { get; set; }
		public IEnumerable<IrrigationEvent> Events { get; set; }
		public int TotalEvents { get; set; }
		public int TotalEventsWithUnknownBearings { get; set; }
		public double PercentageOfZeroBearings { get; set; }
		public IEnumerable<IrrigationEventBoundary> Boundaries { get; set; }

		public IrrigationEventsResponse()
		{
			Summary = new Collection<IrrigationEventSummary>();
			Events = new Collection<IrrigationEvent>();
		}
	}
}

File: reporting-test-client/testClientWebUi/src/app/core/services/irrigation-events-response.ts
import { IrrigationEvent } from './irrigation-event';
import { IrrigationEventSummary } from './irrigation-event-summary';
import { IrrigationEventBoundary } from './irrigation-event-boundary';

export class IrrigationEventResponse {
    summary: IrrigationEventSummary[] = [];
    events: any[] = [];
    boundaries: IrrigationEventBoundary[] = [];
    totalEvents: number;
    totalEventsWithUnknownBearings: number;
    percentageOfZeroBearings: number;
}
